### ISSUE TITLE
[shell] bugs and feedback

### ISSUE THREAD

**Issue by richburdon** [dxos/dxos#4027]
## P0
- [x] Click to copy pin.
- [x] Can't remove pending or "Failed" invitations (no "X").
- [x] Do not show succeeded invitations in the list
- [x] Hangs if attempt to join own space or space you're already a member of (throws exception and dialog doesn't recover). #1922 
- [x] Clicking the "X" on "User Joined" doesn't do anything and throws an error in the console.
- [ ] Create device invitation throws exception: `[vault] invalid invitation {invitationId: 'af0987158a3741b6a2777219d80b1164def8c16dcdfd80927944c44934ee2893'}`

## P1
- [ ] Show error reason (not failed) when invited to a space that the user already belongs to (see image).
- [ ] [composer] No way to cancel a invitation  (i.e., when pin is showing on the sender side).
- [ ] Invitation doesn't timeout (ideal to show countdown).
- [ ] No mechanism to create multi-use or pinless invitations (shift/meta key?)
- [x] #4045 

## P2
- [x] Need headers to delineate between Members and Invites.
- [ ] Show joined peer rather than "User joined" (which appears as an additional item).
- [ ] Auto-copy invite to clipboard when creating invitations.
- [ ] Username overruns end of line (show ellipsis).
- [ ] "Incorrect PIN" label shouldn't shift-up PIN box (just appear below)
- [ ] Clicking "cancel" (on receiver) shows "Failed" (should close dialog).
- [ ] Ideal for demos if the min height could be responsive down to 3 rows.
- [ ] "Copied" popovers add visual noise (not helpful).

## P3
- [ ] Show device list when complete device invitation [IMPORTANT]. #1823 

## P3, Needs more clarity
- [ ] Device panel takes up 1/2 height of screen.

<br/>

![Image](https://user-images.githubusercontent.com/3523355/264184326-e8afcf96-a874-4656-9a93-6fb7ea8b3666.png)

**Comment by jessmartin** [comment]
@richburdon thanks for adding all of these in one place! I added a few items Z and I discussed yesterday. However, the issue is now too big to be a single issue, and clearly not _everything_ on it is a P0 relative to other P0 items in the Strange Loop release.

@zhenyasav let's get together today and divide this into a "now" and "later" issue (and maybe multiple "laters") and prioritize given the larger context of P0 for Strange Loop.
